### PR TITLE
Adding pprof extension

### PR DESCRIPTION
**Pull request by ypatel-crest**
Adding Prof extension

Testing:

- Check with nordport and headless. 
- Check with enabled and disabled pprof.
- Check with multiple nodes.

### CODE REVIEW

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 1]
```diff
@@ -20,6 +20,9 @@ extensions:
     node: ${K8S_NODE_NAME}
 
   zpages:
+  {{- if .Values.agent.pprofExtension }}
+  pprof:
```
[the remark this answers is not in the document]
Comment (ypatel-crest) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:24]: Yes, I have changed it.

**Review by dmitryax — CHANGES_REQUESTED** [review]
As @rmfitzpatrick mentioned, it's already possible to enable pprof with the following config:

```yaml
agent:
  config:
    extensions:
      pprof:
    service:
      extensions:
        - file_storage
        - health_check
        - k8s_observer
        - memory_ballast
        - zpages
        - pprof
```

The values.yaml is already complicated a lot. We don't want to add this kind of configuration for every config customization use case

**Comment by dmitryax** [comment]
I'd would say let's add it in https://github.com/signalfx/splunk-otel-collector-chart/blob/main/examples instead

**Comment by ypatel-crest** [comment]
> I'd would say let's add it in https://github.com/signalfx/splunk-otel-collector-chart/blob/main/examples instead

Yes, I have created a new file in  /example.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**